### ISSUE TITLE
fmt: breaks typedef of functions

### ISSUE THREAD

**Issue by helto4real** [vlang/v#6040]
**V version:** 
V 0.1.28 8dcc3cd

**OS:** 
Ubuntu 18.04
**What did you do?**
```V
pub type SocketMessageFn = fn (mut c Client, msg &Message)?
```
run the fmt

**What did you expect to see?**
A well formatted code.
 
**What did you see instead?**
```V
pub type SocketMessageFn = fn (cmut  Client, msg &Message)
```

**Comment by helto4real** [comment]
V 0.1.28 f9d241a
still broken for me in latest V
@Delta456

**Comment by spytheman** [comment]
it needs to not `mut c` => `cmut` too; the PR fixed only the eating of `?` at the end.

**Comment by spytheman** [comment]
Fixed in  cd9bb51
